### PR TITLE
Migrate to typescript

### PR DESCRIPTION
**Pull request by IlievaDayana**
Work with the incoming data from the socket server;
Migrate all components and state to typescript;
Fix tests;

### CODE REVIEW

File: src/middleware/socket.js [hunk 1]
```diff
@@ -232,81 +180,86 @@ const chatMiddleware = store => next => action => {
...
-    }
+  socket.on(events.streamData, (data) => {
+    const { chat } = store.getState();
 
-    if (chunk.includes('[')) {
-      store.dispatch(setTextToParse(chunk));
-      return;
+    const current = mockedDataForEachStream[idx];
+    if (current) {
+      if (current.type === 'email') store.dispatch(setIsEmailFormVisible(true));
```
Comment (venzovisa) [inline src/middleware/socket.js]: @IlievaDayana It is better to be consistent and always use curly braces for if statement body like the format used on line 237 below.

`if (current.type === 'email') {
   store.dispatch(setIsEmailFormVisible(true));
}`

File: src/middleware/socket.js [hunk 2]
```diff
@@ -219,7 +168,6 @@ const chatMiddleware = store => next => action => {
 
   socket.on(events.chatHistory, ({ history, errors, region }) => {
     store.dispatch(resetIsLoading());
-    store.dispatch(resetIncoming());
     store.dispatch(setRegion(region));
 
     const { config, meta, chat } = store.getState();
```
Comment (venzovisa) [inline src/middleware/socket.js]: @IlievaDayana You could destruct 

`const { config, meta: { cid, region }, chat }`

and use it below:

`
handleMessageSending({
      user_id: cid,
      region
 });
`

**Review by mitevaAlex — APPROVED** [review]

**Review by venzovisa — APPROVED** [review]

File: src/interfaces/index.ts [hunk 3]
```diff
@@ -80,6 +80,55 @@ export interface ChatState {
...
+  messaging: {
+    isVisible: boolean
+  },
+  link: {
+    isVisible: boolean,
+    href: string,
+    text: string
+  }
+}
 export interface UserHistoryData {
```
Comment (venzovisa) [inline src/interfaces/index.ts:109]: @IlievaDayana I think a better approach is to separate nested properties as separate types like these even though there are not exported anywhere else in the app. It improves readability: 

`interface IntentionsStateEmail {
   current: string,
    success: boolean,
    error: boolean,
    isFormVisible: boolean,
    isLoading: boolean,
}

export interface IntentionsState {
  email: IntentionsStateEmail,
  response: IntentionsStateResponse
  ...
}`

File: src/config/index.ts [hunk 4]
```diff
@@ -237,37 +237,69 @@ export const serverHistoryMockWithPaymentIntent = [
...
+    if ('payment' in initial && 'payment' in current) {
+      return {
+        ...initial,
+        ...current,
+        payment: initial.payment + current.payment
+      } as PaymentMessage
+    }
+    return initial
+  },
 };
```
[the remark this answers is not in the document]
Comment (IlievaDayana) [reply src/config/index.ts:120]: i think this is alredy redone

**Pull request merged**